RxCocoa/Common/DelegateProxyType.swift

```diff
@@ -210,6 +213,14 @@ extension DelegateProxyType {
             proxy.setForwardToDelegate(nil, retainDelegate: retainDelegate)
         }
     }
+    
+    public static func extend<Object: AnyObject>(with factory: @escaping ((Object) -> AnyObject)) {
+        factories.append({ ($0 as? Object).map(factory) })
```

Comment:
Please take a look at `createProxyForObject` method suggestion first.

We need to crash the app in case extend is called from non main scheduler. // thread safety
We need to crash the app in case somebody registers extension for the same class twice.